### PR TITLE
Add base package structure

### PR DESCRIPTION
**Pull request by jni**
This adds most of the base structure needed for the package. We can edit as we go.

- Add setup.cfg
- Add basic package structure
- Add docs folder
- Add test and docs deployment on GHA

### CODE REVIEW

File: .github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      # these libraries enable testing on Qt on linux
+      - uses: tlambert03/setup-qt-libs@v1
```
Comment (JoOkuma) [inline .github/workflows/test_and_deploy.yml:37]: Do we need the `setup-qt-libs`?

File: docs/api/api.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Skan API Reference
```
Comment (JoOkuma) [inline docs/api/api.md:1]: Iskan -> napari-graph

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+project = 'Skan'
+copyright = '2021, Juan Nunez-Iglesias & skan contributors'
+author = 'Juan Nunez-Iglesias'
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+import skan
```
Comment (JoOkuma) [inline docs/conf.py:75]: you're importing the `skan` package and you mention it in some places.

**Comment by JoOkuma** [comment]
I couldn't run it because of missing `setup.py`.
I can add this later when we add the first graph class.

**Comment by jni** [comment]
Oops! Thanks for the review! 😅 Fixing now.

> I couldn't run it because of missing setup.py.

You *should* be able to `pip install .` with just a setup.cfg these days? Checking...

**Comment by jni** [comment]
Ah, nope, need `pyproject.toml`. I'll get this right one of these days... 😂

**Comment by tlambert03** [comment]
You'll need more than an empty pyproject.  It needs the build system:  https://setuptools.pypa.io/en/latest/build_meta.html#how-to-use-it

**Comment by jni** [comment]
Thanks for the link @tlambert03 ! `pip install .` worked with the empty toml so I thought that was enough. 😅 Will update.

**Comment by JoOkuma** [comment]
Looks good to me! It's also working here with an empty pyproject, even with some function inside the `__init__.py`.

**Pull request merged**